lib/shopify-cli/commands/authenticate.rb

```diff
@@ -6,19 +6,28 @@ class Authenticate < ShopifyCli::Command
       def call(args, _name)
         command = args.shift
         ShopifyCli::Tasks::EnsureLoopbackURL.call(@ctx)
-        CLI::UI::Spinner.spin("Requesting access token...") do |spinner|
-          begin
-            case command
-            when 'identity'
-              ShopifyCli::Tasks::AuthenticateIdentity.call(@ctx)
-            else
-              ShopifyCli::Tasks::AuthenticateShopify.call(@ctx)
-            end
-            spinner.update_title("Authetication token stored")
-          rescue OAuth::Error
-            @ctx.puts("{{error:Failed to Authenticate}}")
-            raise(::ShopifyCli::Abort, "Failed to Authenticate")
+        begin
+          case command
+          when 'identity'
+            run_task(ShopifyCli::Tasks::AuthenticateIdentity.call(@ctx))
+          else
```

Comment:
Slight modification, I think we should have this be an explicit `shop` case/subcommand and the else clause just show help.

Reply:
okay! so the use can call `identity` or `shop`?

Reply:
yup!